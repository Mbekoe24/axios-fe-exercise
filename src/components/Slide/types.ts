export interface ApiResponse extends Omit<SlideProps, "htmlId">{
  id: string
}
export interface SlideProps {
  headline: string,
  htmlId: string,
  displayName: string,
  sectionLabel: string,
  permaLink: string, 
  publishedDate: string, 
  primaryImage: string
}