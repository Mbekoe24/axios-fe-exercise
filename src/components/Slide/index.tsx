import React from "react";
import { SlideProps } from "./types";

export const Slide = (props: SlideProps) => {
  const formattedDate = new Date(props.publishedDate).toLocaleDateString(
    "en-US",
    {
      dateStyle: "long",
    }
  );

  return (
    <div id={props.htmlId}>
      <a href={props.permaLink} target="_blank" rel="noopener noreferrer">
        <img src={props.primaryImage} alt={props.altText} />
      </a>{" "}
      <div className="info-box">
        <p>{props.headline}</p>
        <span>
          <time>{formattedDate}</time> - {props.topicName}
          {""}
        </span>
      </div>
    </div>
  );
};
