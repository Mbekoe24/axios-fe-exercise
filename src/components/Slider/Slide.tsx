import React from "react";
import { SlideProps } from "../../../types";

export const Slide = (props: SlideProps) => {
  return (

    ////// testing if data populates
    <div className="slides">
      <div id="slide-1">
        <a href={props.permaLink}>
          <img src={props.primaryImage} alt="" />
        </a>
        <h1>{props.headline}</h1>
        <span>
          {props.publishedDate} - {props.sectionLabel}
        </span>
      </div>
      <div id="slide-2">2</div>
      <div id="slide-3">3</div>
      <div id="slide-4">4</div>
      <div id="slide-5">5</div>
    </div>
  );
};
