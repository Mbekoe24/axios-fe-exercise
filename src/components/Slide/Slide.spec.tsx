import { fireEvent, render, screen } from "@testing-library/react";
import user from "@testing-library/user-event";
import * as React from "react";
import Slide from ".";

const mockData = {
  key: "57a186a3-7547-45bf-852a-83019849d8d5",
  htmlId: "slide-2",
  headline: "The gamification of gambling",
  displayName: "Pete Gannon",
  topicName: "Economy & Business",
  altText:
    "Illustration of a hand of high cards in five separate mobile phones. ",
  publishedDate: "2022-06-28T21:13:23.963000Z",
  primaryImage:
    "https://images.axios.com/qM2XadizW4NmxcXRDhDPZg7ZAtQ=/fit-in/1366x1366/2022/06/28/1656438051676.jpg",
  permaLink: "https://www.axios.com/2022/06/28/the-gamification-of-gambling",
};

describe("<Slide />", () => {
  it("Should render correctly", () => {
    const { container } = render(<Slide {...mockData} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  describe("Slide component should have an image", () => {
    test("Should render image screen", () => {
      render(<Slide {...mockData} />);

      const image = screen.getByTestId("story-img");
      expect(image).toBeInTheDocument();
    });
  });

  describe("Slide component should have a link on the image", () => {
    test("Should render a link on the image ", () => {
      render(<Slide {...mockData} />);

      const link = screen.getByTestId("story-link");
      expect(link).toBeInTheDocument();
      expect(link).toContainHTML("img");
      user.click(link);
      expect(link).toBeEnabled();
    });
  });
});
