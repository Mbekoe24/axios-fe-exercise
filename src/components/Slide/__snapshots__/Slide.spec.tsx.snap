// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slide /> Should render correctly 1`] = `
<div
  id="slide-2"
>
  <a
    data-testid="story-link"
    href="https://www.axios.com/2022/06/28/the-gamification-of-gambling"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="Illustration of a hand of high cards in five separate mobile phones. "
      data-testid="story-img"
      src="https://images.axios.com/qM2XadizW4NmxcXRDhDPZg7ZAtQ=/fit-in/1366x1366/2022/06/28/1656438051676.jpg"
    />
  </a>
  <div
    class="info-box"
  >
    <p>
      The gamification of gambling
    </p>
    <span>
      <time>
        6/28/2022
      </time>
       - 
      Economy & Business
    </span>
  </div>
</div>
`;
